Managers/UInputMan.cpp

```diff
@@ -939,172 +981,180 @@ namespace RTE {
-				int whichJoy = GetJoystickIndex(device);
-				if (whichJoy == joystick) {
+
+			for (int playerToCheck = Players::PlayerOne; playerToCheck < Players::MaxPlayerCount; ++playerToCheck) {
+				InputDevice playerInputDevice = m_ControlScheme[playerToCheck].GetDevice();
+				int whichJoy = GetJoystickIndex(playerInputDevice);
+				if (whichJoy == joystickIndex) {
 					deadZone = m_ControlScheme[playerToCheck].GetJoystickDeadzone();
 					deadZoneType = m_ControlScheme[playerToCheck].GetJoystickDeadzoneType();
 					joystickPlayer = static_cast<Players>(playerToCheck);
-					break;
 				}
 			}
-			if (joystickPlayer > Players::NoPlayer && deadZoneType == DeadZoneType::CIRCLE && deadZone > 0.0F) {
+
+			bool isAxisMapped = false;
+			if (joystickPlayer != Players::NoPlayer && deadZone > 0.0F) {
 				Vector aimValues;
 				const std::array<InputMapping, InputElements::INPUT_COUNT> *inputElements = m_ControlScheme[joystickPlayer].GetInputMappings();
-				std::array<InputElements, 4> elementsToCheck = { InputElements::INPUT_L_LEFT, InputElements::INPUT_L_UP, InputElements::INPUT_R_LEFT, InputElements::INPUT_R_UP };
-
-				for (int i = 0; i < elementsToCheck.size() - 1; i += 2) {
-					if (inputElements->at(elementsToCheck[i]).JoyDirMapped()) { aimValues.m_X = AnalogAxisValue(joystick, inputElements->at(elementsToCheck[i]).GetStick(), inputElements->at(elementsToCheck[i]).GetAxis()); }
-					if (inputElements->at(elementsToCheck[i + 1]).JoyDirMapped()) { aimValues.m_Y = AnalogAxisValue(joystick, inputElements->at(elementsToCheck[i + 1]).GetStick(), inputElements->at(elementsToCheck[i + 1]).GetAxis()); }
-
-					if (aimValues.MagnitudeIsLessThan(deadZone * 2.0F)) {
-						for (int j = 0; j < 2; j++) {
-							InputElements whichElementDirection = elementsToCheck[i + j];
-							if (inputElements->at(whichElementDirection).JoyDirMapped()) {
-								JOYSTICK_AXIS_INFO *joystickAxis = &joy[joystick].stick[inputElements->at(whichElementDirection).GetStick()].axis[inputElements->at(whichElementDirection).GetAxis()];
-								if (joy[joystick].stick[inputElements->at(whichElementDirection).GetStick()].flags & JOYFLAG_UNSIGNED) {
-									joystickAxis->pos = 128;
-									joystickAxis->d1 = joystickAxis->d2 = 0;
-								} else {
-									joystickAxis->pos = joystickAxis->d1 = joystickAxis->d2 = 0;
-								}
-							}
-						}
+				std::array<InputElements, 4> elementsToCheck = {InputElements::INPUT_L_LEFT, InputElements::INPUT_L_UP, InputElements::INPUT_R_LEFT, InputElements::INPUT_R_UP};
+
+				for (std::size_t i = 0; i < elementsToCheck.size(); i += 2) {
+					int axisLeft{SDL_CONTROLLER_AXIS_INVALID};
+					int axisUp{SDL_CONTROLLER_AXIS_INVALID};
+					if (inputElements->at(elementsToCheck[i]).JoyDirMapped()) {
+						axisLeft = inputElements->at(elementsToCheck[i]).GetAxis();
+						aimValues.m_X = AnalogAxisValue(joystickIndex, axisLeft);
+						isAxisMapped |= axisLeft == axis;
```

Comment:
I'd rather not use `|=` here. I know it works, but it's bitwise or and it's not really for this.